client/signup/main.jsx

```diff
@@ -149,7 +151,14 @@ const Signup = React.createClass( {
 			defer( function() {
 				page( destination );
 			}.bind( this ) );
-		} else {
+		}
+
+		if ( ! user.get() && config.isEnabled( 'oauth' ) ) {
```

Comment:
Ever since I saw [this](https://www.youtube.com/watch?v=XDqZemwVXUk&feature=youtu.be&t=16m42s) video, I've preferred to avoid `else` and nested conditions in favor of a flat list of `if` statements, because I find the latter more readable/scalable. I could be in the minority on this one though. :sweat_smile: